travis/update-release.py

```diff
@@ -179,66 +205,38 @@ def check_rollback_artifacts(r_stream,tag):
             # Change the working directory temporarily
             os.chdir("/tmp/" + GIT_LOCAL_DIR)
 
-            # Command 1: git stash
-            subprocess.run(["git", "stash"], universal_newlines=True)
+            # Run the git workflow
+            run_git_commands(quaySha)
 
-            # Command 2: git log --grep=xyz --format=%H
-            git_log_command = ["git", "log", "--grep=" + quaySha, "--format=%H"]
-            commit_hash = subprocess.check_output(git_log_command, universal_newlines=True).strip()
+            # Return to the original working directory
+            os.chdir(current_directory)
 
-            # Command 3: git checkout abcd
-            subprocess.run(["git", "checkout", commit_hash], universal_newlines=True)
+            image_update = create_release_image_data(image, dockerSha,quaySha,tag)
 
-            # Command 4: copy some files to a destination (e.g., using shutil or any other method)
-            copyfile("docs/release_artifacts/" + RELEASE_TAG + "/z/" + image["name"],
-                            "/tmp/z/"+image["name"])
+            c_images.append(image_update)
+
+        return c_images, True
+    elif new_sync:
+        print("Syncing to new images ...")
+        for image in z_container_images:
+            # lookup image sha's
+            quaySha = pull_image_and_get_sha("quay.io/noiro/" + image["name"] + ":" + tag)
+            # lookup image sha
+            dockerSha = pull_image_and_get_sha("noiro/" + image["name"] + ":" + tag)
 
-            # Command 5: git checkout main
-            subprocess.run(["git","checkout", "main"], universal_newlines=True)
+            current_directory = os.getcwd()
 
-            # Command 6: git stash pop
-            subprocess.run(["git","stash", "pop"], universal_newlines=True)
+            # Change the working directory temporarily
+            os.chdir("/tmp/" + GIT_LOCAL_DIR)
 
-            copyfile("/tmp/z/" + image["name"], "docs/release_artifacts/" + RELEASE_TAG + DIR + image["name"])
+            # Command 4: copy z files to a release (e.g., using shutil or any other method)
+            copyfile("docs/release_artifacts/" + RELEASE_TAG + "/z/" + image["name"],
```

Comment:
new-sync would be called for docker rc/release images.
maybe in this case simply copying the artifacts is not sufficient or accurate, while we cannot provide build-logs for the image who's sha doesn't exist, I think we can provide accurate or new cve or sbom for it, since if we only copy the previous data, it is equvalent to the release being copied over from latest 'z' release, which may not be correct, if the image was built from an older container job.

Reply:
Again, the limitation here is that we pushed release image to docker from the latest z tag for which we have release artifacts. The thinking is right, but seeing the CICD workflow, 2 things 
1. : The rollback only happens if we tagged the quay image to older one as in quay only we can tag an image without changing to new-sha, it can take older sha for which we have history
2. For docker it will always be a new sha , which we will assume that we do for .z. If now we do for older image, new-sha is generated, in that case we are pointing to artifacts of z in the assumption that no major changes between between older tag and z tag and we provide the cve sbom for those builds. 

Checking on this one as well.